tests/cpp/dlr_tvm_test.cc

```diff
@@ -1,8 +1,136 @@
 #include <gtest/gtest.h>
+#include "dlr_tvm.h"
+#include "test_utils.hpp"
 
-#include "dlr.h"
 
-TEST(TVM, Test1) { EXPECT_EQ(1, 1); }
+TEST(TVM, TestTvmModelApisWithOutputMetadata) {
+  const int device_type = 1;  // 1 - kDLCPU
+  const int device_id = 0;
+  DLContext ctx  = {
+    static_cast<DLDeviceType>(device_type),
+    device_id
+  };
+  std::vector<std::string> paths = {"./resnet_v1_5_50"};
+  dlr::TVMModel model = dlr::TVMModel(paths, ctx);
+  
+  EXPECT_EQ(model.GetNumInputs(), 1);
+  EXPECT_EQ(model.GetNumOutputs(), 2);
+  EXPECT_EQ(model.HasMetadata(), true);
+  EXPECT_STREQ(model.GetOutputName(0), "ArgMax:0");
+  EXPECT_STREQ(model.GetOutputType(0), "int32");
+  EXPECT_STREQ(model.GetOutputName(1), "softmax_tensor:0");
+  EXPECT_STREQ(model.GetOutputType(1), "float32");
+
+  const int batch_size = 1;
+  size_t img_size = 224*224*3;
+  float* img = LoadImageAndPreprocess("cat224-3.txt", img_size, batch_size);
+  img_size *= batch_size;
+  int64_t shape[4] = {1, 224, 224, 3};
+
+  int64_t output_size;
+  int output_dim;
+  int index = 0;
+  EXPECT_NO_THROW(model.GetOutputSizeDim(index, &output_size, &output_dim));
+  
+  EXPECT_NO_THROW(model.SetInput("input_tensor", shape, img, 4));
+  EXPECT_NO_THROW(model.Run());
+
+  float* output1[output_size];
+  float* output2[output_size];
+  EXPECT_NO_THROW(model.GetOutput(index, output1));
+  EXPECT_NO_THROW(model.GetOutputByName(model.GetOutputName(index), output2));
+  EXPECT_EQ(output1[0], output2[0]);
+
+  try {
+    model.GetOutputIndex("blah");
+  } catch(const dmlc::Error& e) {
+    EXPECT_STREQ(e.what(), "Couldn't find index for output node!");
+  }
+  
+  try {
+    model.GetOutputByName("blah", output1);
+  } catch(const dmlc::Error& e) {
+    EXPECT_STREQ(e.what(), "Couldn't find index for output node!");
+  }
+}
```

Comment:
Can you also test GetOutput() where index is greater than the number of outputs?

Reply:
👍